### PR TITLE
Convert fizz buzz to oo

### PR DESCRIPTION
**Pull request by katemildner**

### CODE REVIEW

**Review by ktopolski — COMMENTED** [review]
Hello Kate,

I don't really know if it's a WIP PR or is it considered done, so I'm commenting without explicitly approving or rejecting the changes.

I love the new directory tree, imho it's a lot cleaner than previous version with long file names. As you know, the O letters put next to each other makes me smile, so I definitely like the direction you're heading. However, I'm having trouble getting where you're coming from.

I'm a little worried that the final destination is unit testing UI and as much as I support that, it might turn out to be a rabbit hole with `StringIO` waiting at the end. I very much welcome the UI unit tests for parsing the given argument via `begin/rescue` block or transforming the argument with things like `gsub`. Just please be careful when it will come to test things like `puts` and `gets`.

Testing `puts` and `gets` touches subjects like input and output streams and dependency injection. As much as I believe knowledge of those two is essential for a full-stack developer, it might just be too much of a hassle at the beginning of your journey.

As always it's your journey and if you decide to become Alice in the Streamland I'd happily take a role of a guide as usual :).

Cheers,
Karol

File: FizzBuzz/UI - multiple parameters.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  def fizz_buzz_input(input, fb_case)
+    begin
+      input = Float(input)
+    rescue ArgumentError => e
+      puts "#{input} is an improper value: #{e.class.name}, try again."
+      is_float = false
+    end
+
+    if is_float == false
```
Comment (ktopolski) [inline FizzBuzz/UI - multiple parameters.rb]: it doesn't seem like we're doing much in those empty `if` blocks like this one or `input%1 == 0` below, so might be a good idea to get rid of them.

**Comment by katemildner** [comment]
Hi,

Thanks for your feedback - I was somehow convinced writing tests for UI won't be a bother at all, but the more I try to get a grip of it, the more I see how much I don't know yet. I'll definitely get back to this, but I'll pass on the chase after the white rabbit for now - just like you said, It'll be too much of a hassle at the moment.
Nonetheless, I couldn't leave UI in its original shape, so at least that's that. :)

File: FizzBuzz/UI - multiple parameters.rb (hunk 1 above)
Comment (katemildner) [reply FizzBuzz/UI - multiple parameters.rb]: I'm not sure if I know how to do it without crippling the whole thing, but I'm on it. Thanks!

File: FizzBuzz/logic.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Fizz_Buzz
```
Comment (ktopolski) [inline FizzBuzz/logic.rb]: underscores in class names in ruby?
obraz

please change it to camel case :)

Comment (ktopolski) [reply FizzBuzz/logic.rb]: also - in general we should stick to best practices of naming the file after the class the file defines. So, here we're aiming more for `FizzBuzz/fizz_buzz.rb` and `class FizzBuzz`

File: FizzBuzz/UI - single parameter.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    begin
+      input = Float(input)
+    rescue ArgumentError => e
+      puts "Improper value: #{e.class.name}, try again."
+      @fb_case.print_fizz_buzz(input)
+      exit
+    end
+
+    if input%1 == 0
```
Comment (ktopolski) [inline FizzBuzz/UI - single parameter.rb]: better way would be just:

```ruby
if input%1 != 0
  puts "Would you like to round your number #{input}? Y/N"
```

or, more ruby syntax:

```ruby
unless input%1 == 0
```

having empty `if` condition does not contribute much :).

File: FizzBuzz/UI - multiple parameters.rb (hunk 1 above)
Comment (ktopolski) [reply FizzBuzz/UI - multiple parameters.rb]: just check if `is_float` is true instead :)

**Review by ktopolski — CHANGES_REQUESTED** [review]
Hello Kate,

I agree with your decision of abandoning the Streamland rabbit hole. There's always a time to come back to this topic after gaining more experience.

The main fix of this PR is going to be filenames and class names. In ruby we use CamelCase for class names and snake_case for files, where file should be named after the class it contains.

I also dropped few hints on empty conditionals :). Apart from that - really nice, I also appreciate commit messages getting better.

Looking forward to the merge!
Karol

File: FizzBuzz/ui_multiple_parameters.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  def fizz_buzz_input(input, fb_case)
+    begin
+      input = Float(input)
+    rescue ArgumentError => e
+      puts "#{input} is an improper value: #{e.class.name}, try again."
+      is_float = false
+    end
+
+    if is_float == false
```
Comment (ktopolski) [inline FizzBuzz/ui_multiple_parameters.rb]: stil an empty one :)

**Review by ktopolski — APPROVED** [review]

**Comment by ktopolski** [comment]
Took some time, but finally made it!

Happy to merge this one :).

**Pull request merged**